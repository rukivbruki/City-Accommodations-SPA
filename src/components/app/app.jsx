import React from 'react';
import {Router, Switch, Route} from 'react-router-dom';
import ScrollToTop from '../scroll-to-top/scroll-to-top.jsx';
import history from '../../utils/history';

import MainPage from '../main-page/main-page.jsx';
import SignInPage from '../sign-in-page/sign-in-page.jsx';
import Header from '../header/header.jsx';
import FavoritesPage from '../favorites-page/favorites-page.jsx';
import PlacePage from '../place-page/place-page.jsx';

const App = () => {
  return <>
    <Router history={history}>
      <ScrollToTop>
        <Header />
        <Switch>
          <Route path="/" exact component={MainPage}/>
          <Route path="/login" component={SignInPage} />
          <Route path="/favorites" component={FavoritesPage} />
          <Route path="/offer/:id" component={PlacePage} />
        </Switch>
      </ScrollToTop>
    </Router>
  </>;
};

export {App};
export default App;


