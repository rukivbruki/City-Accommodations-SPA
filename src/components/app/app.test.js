import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {App} from './app.jsx';

it(`App snapshot`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<App/>);
  expect(tree).toMatchSnapshot();
});
