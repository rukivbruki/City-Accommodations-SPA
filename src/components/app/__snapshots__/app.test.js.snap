// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<React.Fragment>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ScrollToTop)>
      <Memo(ConnectFunction) />
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(WithActivePlace)",
              "type": [Function],
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(WithForm)",
              "type": [Function],
            }
          }
          path="/login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Connect(WithAuthorizationReqired))",
              "type": [Function],
            }
          }
          path="/favorites"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PlacePage)",
              "type": [Function],
            }
          }
          path="/offer/:id"
        />
      </Switch>
    </withRouter(ScrollToTop)>
  </Router>
</React.Fragment>
`;
